src/dbnode/client/client.go

```diff
@@ -95,15 +95,19 @@ func (c *client) Options() Options {
 }
 
 func (c *client) NewSession() (Session, error) {
-	return c.newSession()
+	return c.newSession(c.opts)
+}
+
+func (c *client) NewSessionWithOpts(opts Options) (Session, error) {
```

Comment:
not: NewSessionWithOptions?

Opts and other abbreviations we tend to use with local vars not exported/public APIs. Definitely a nit though.